python/lsst/ap/association/association.py

```diff
@@ -405,6 +406,36 @@ def _check_dia_object_position(self, dia_object_record, bbox, wcs):
                                                 geom.degrees))
         return bbox.contains(point)
 
+    def check_dia_souce_radec(self, dia_sources):
+        """Check that all DiaSources have non-NaN values for RA/DEC.
+
+        If one or more DiaSources are found to have NaN values, throw a
+        warning to the log with the ids of the offending sources. Drop them
+        from the table.
+
+        Parameters
+        ----------
+        dia_sources : `pandas.DataFrame`
+            Input DiaSources to check for NaN values.
+
+        Returns
+        -------
+        trimmed_sources : `pandas.DataFrame`
+            DataFrame of DiaSources trimmed of all entries with NaN values for
+            RA/DEC.
+        """
+        nan_mask = np.logical_or(np.isnan(dia_sources.loc[:, "ra"]),
+                                 np.isnan(dia_sources.loc[:, "decl"]))
+        if np.any(nan_mask):
+            nan_idxs = np.argwhere(nan_mask).flatten()
+            for nan_idx in nan_idxs:
+                self.log.warning(
+                    "DiaSource %i has NaN value for RA/DEC, "
+                    "dropping from association." %
+                    dia_sources.loc[nan_idx, "diaSourceId"])
+            dia_sources = dia_sources[np.logical_not(nan_mask)]
```

Comment:
You might try a timing test of just this line and see if `dia_sources[~nan_mask]` is faster.

Reply:
I doubt there is much difference here so I'll change it to the simpler line.